.github/workflows/mpc-test-ci.yaml

```diff
@@ -0,0 +1,66 @@
+name: Validate PR - Testing Phase - golang CI
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    branches:
+      - main
+jobs:
+  checkout_and_test:
+    name: Golang Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code of the forked repository's head branch
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}  
+          ref: ${{ github.event.pull_request.head.ref }}  
+      - name: Install Go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5
+        with:
+          go-version-file: "./go.mod"
+      - name: Download SeaLights Go agent and CLI tool
+        run: |
+          echo "[Sealights] Downloading Sealights Golang & CLI Agents..."
+          case $(lscpu | awk '/Architecture:/{print $2}') in
+            x86_64) SL_ARCH="linux-amd64";;
+            arm) SL_ARCH="linux-arm64";;
+          esac
+          wget -nv -O sealights-go-agent.tar.gz https://agents.sealights.co/slgoagent/latest/slgoagent-$SL_ARCH.tar.gz
+          wget -nv -O sealights-slcli.tar.gz https://agents.sealights.co/slcli/latest/slcli-$SL_ARCH.tar.gz
+          tar -xzf ./sealights-go-agent.tar.gz && tar -xzf ./sealights-slcli.tar.gz
+          rm -f ./sealights-go-agent.tar.gz ./sealights-slcli.tar.gz
+          ./slgoagent -v 2> /dev/null | grep version && ./slcli -v 2> /dev/null | grep version
```

Comment:
Consider caching SeaLights so its not downloaded on every job run

Reply:
I asked Marc (the tech-sales guy from SeaLights) how I can check the version of the agent remotely so I can do this. We have a meeting tomorrow with SeaLights about a better solution for fetching the agent, if he won't answer me today, I'll ask him on the meeting.